app/javascript/lib/validation/validate-file-content.js

```diff
@@ -177,9 +177,89 @@ function validateEqualCount(headers, annotTypes) {
   return issues
 }
 
+/**
+ * Verify "GENE" is present as the first column in the first row for an Expression Matrix file
+ * Todo: Accept files that are R-formatted as well via this ticket: SCP-
+ */
+function validateGeneInHeader(headers) {
+  const issues = []
+  if (headers[0][0].toUpperCase() !== 'GENE') {
+    const msg = 'Dense matrices require the first value of the file to be "GENE". ' +
+     `However, the first value for this file currently is "${headers[0][0]}".`
+    issues.push(['error', 'format:cap:missing-gene-column', msg])
+  }
+
+  return issues
+}
+
+
+/**
+ *
+ * Validate all values are numbers outside first column cell name
+ */
+function validateValuesAreNumeric(parsedLine, isLastLine, lineNum, dataObj) {
+  const issues = []
+
+  // skip first column
+  const lineWithoutFirstColumn = parsedLine.slice(1)
+  console.log('parsedLine after', parsedLine)
+
+  if (lineWithoutFirstColumn.some(isNaN)) {
+    dataObj.rowsWithNonNumericValues.add(lineNum)
+  }
+
+  const numBadRows = dataObj.rowsWithNonNumericValues.size
+  if (isLastLine && numBadRows > 0) {
+    const rowText = numBadRows > 1 ? 'rows' : 'row'
+    const max = 10
+    let notedBadRows = dataObj.rowsWithNonNumericValues.join(', ').slice(0, max)
+
+    if (numBadRows - max > 0) {
+      notedBadRows += ` and ${numBadRows - max} more rows`
+    }
+
+    const msg = `All values (other than the first column and header row) in a dense matrix file must be numeric. ` +
+    `Please ensure all values in ${rowText}: ${notedBadRows.join(', ')}, are numbers.`
+    issues.push(['error', 'format:invalid-type:not-numeric', msg])
+  }
+  return issues
+}
+
+
+
+/**
+ * Verify dense matrix column numbers match header column numbers
+ */
+function validateColumnNumber(parsedLine, isLastLine, headers, lineNum, dataObj) {
+  const issues = []
+  // use the first header row to determine the correct number of columns all rows should have
+  const correctNumberOfColumns = headers[0].length
+
+  if (correctNumberOfColumns !== parsedLine.length) {
```

Comment:
Per rationale outlined in my comment beginning "Similar to".

```suggestion
function validateColumnNumber(line, isLastLine, headers, lineNum, dataObj) {
  const issues = []
  // use the first header row to determine the correct number of columns all rows should have
  const correctNumberOfColumns = headers[0].length

  if (correctNumberOfColumns !== line.length) {
```